src/rasmus_fuel/realistic.py

```diff
@@ -95,10 +95,27 @@ def power_maintain_sog(
     v_ship_tw = v_ship_og - v_current
 
     # calc speeds (over ground, through water, relative to wind)
-    speed_og = (u_ship_og ** 2 + v_ship_og ** 2) ** 0.5
+    # speed_og = (u_ship_og ** 2 + v_ship_og ** 2) ** 0.5
     speed_tw = (u_ship_tw ** 2 + v_ship_tw ** 2) ** 0.5
     speed_rel_to_wind = ((u_ship_og - u_wind) ** 2 + (v_ship_og - v_wind) ** 2) ** 0.5
 
+    # calc wind direction and speed
+    angle_wind = np.arctan2(v_wind, u_wind)
+    speed_wind = (u_wind ** 2 + v_wind ** 2) ** 0.5
+
+    # calc angle between wind and speed through water
+    angle_ship_wind = np.where(
+        ((speed_tw > 10e-4) & (speed_wind > 10e-4)),
+        np.arccos(
+            (u_wind * u_ship_tw + v_wind * v_ship_tw) / (speed_tw + 10e-6) / (speed_wind + 10e-6)
+        ),
+        0,
+    )
+
+    # calc projection of ship velocity through water on wind components
+    u_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.cos(angle_wind)
+    v_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.sin(angle_wind)
+
```

Comment:
I think this should just be:
```suggestion
    u_speed_rel_to_wind = u_ship_og - u_wind
    v_speed_rel_to_wind = v_ship_og - v_wind
```

The strength and direction of the drag force is evaluated from the relative movement of the air across the vessel. So we just take the difference between the ship velocity vector and the wind velocity vector in any reference system. As the wind is given over ground, we also use ship velocity over ground.

Note that for the conversion to power, we still need to multiply with speed through water as this is the reference system relevant for the propulsion setup.

Reply:
Yes, clear. For the wind-ship we do use reference with respect to Earth observer. I was confused.